File: api_bus/core/models/trabajador.py
from django.db import models
from .usuario import Usuario

class Trabajador(models.Model):
    PUESTO_CHOICES = [
        ('conductor', 'Conductor'),
        ('supervisor', 'Supervisor'),
        ('mantenimiento', 'Mantenimiento'),
        ('administrativo', 'Administrativo'),
    ]
    
    TURNO_CHOICES = [
        ('matutino', 'Matutino'),
        ('vespertino', 'Vespertino'),
        ('nocturno', 'Nocturno'),
        ('mixto', 'Mixto'),
    ]
    
    usuario = models.OneToOneField(Usuario, on_delete=models.CASCADE, related_name='trabajador')
    puesto = models.CharField(max_length=14, choices=PUESTO_CHOICES)
    turno = models.CharField(max_length=10, choices=TURNO_CHOICES)
    fecha_ingreso = models.DateField()
    
    def __str__(self):
        return f"{self.usuario.persona.nombre} - {self.puesto}"

File: api-v2/models/trabajador.js
import { DataTypes } from "sequelize";
import sequelize from "../config/db.js";

const PUESTO = ["conductor", "supervisor", "mantenimiento", "administrativo"];
const TURNO = ["matutino", "vespertino", "nocturno", "mixto"];

const Trabajador = sequelize.define(
  "Trabajador",
  {
    id: {
      type: DataTypes.BIGINT.UNSIGNED,
      autoIncrement: true,
      primaryKey: true,
    },
    usuarioId: {
      type: DataTypes.BIGINT.UNSIGNED,
      allowNull: true,
      unique: true,
      field: "usuario_id", // 👈 asegura correspondencia con la columna
      references: { model: "Usuarios", key: "id" },
      onUpdate: "CASCADE",
      onDelete: "SET NULL",
    },
    puesto: {
      type: DataTypes.STRING(32),
      allowNull: true,
      validate: { isIn: [PUESTO] },
    },
    turno: {
      type: DataTypes.STRING(16),
      allowNull: true,
      validate: { isIn: [TURNO] },
    },
    fecha_ingreso: { type: DataTypes.DATEONLY, allowNull: true },
  },
  {
    tableName: "Trabajadores",
    timestamps: true,
    underscored: true,
  }
);

Trabajador.associate = (models) => {
  Trabajador.belongsTo(models.Usuario, {
    as: "usuario",
    foreignKey: "usuarioId",
  });
};

Trabajador.addHook("beforeSave", async (t) => {
  if (!t.usuarioId) return;
  const { Usuario, Persona } = sequelize.models;
  const u = await Usuario.findByPk(t.usuarioId, { attributes: ["personaId"] });
  if (!u) throw new Error("El usuario asociado no existe.");
  if (!u.personaId) throw new Error("El usuario no tiene persona asociada.");
  const p = await Persona.findByPk(u.personaId, { attributes: ["tipo"] });
  if (!p || p.tipo !== "trabajador") {
    throw new Error(
      'La Persona asociada al usuario debe tener tipo "trabajador".'
    );
  }
});

export default Trabajador;
